integration-tests/src/test/java/org/apache/fineract/integrationtests/SchedulerJobsTestResults.java

```diff
@@ -376,6 +376,86 @@ public void testApplyHolidaysToLoansJobOutcome() throws InterruptedException {
+        loanStatusHashMap = this.loanTransactionHelper.disburseLoanWithNetDisbursalAmount(AccountTransferTest.LOAN_DISBURSAL_DATE, loanID,
+                JsonPath.from(loanDetails).get("netDisbursalAmount").toString());
+        LoanStatusChecker.verifyLoanIsActive(loanStatusHashMap);
+
+        // Retrieving All Global Configuration details
+        final ArrayList<HashMap> globalConfig = GlobalConfigurationHelper.getAllGlobalConfigurations(requestSpec, responseSpec);
+        Assertions.assertNotNull(globalConfig);
+
+        // Updating Value for reschedule-repayments-on-holidays Global
+        // Configuration
+        Integer configId = (Integer) globalConfig.get(3).get("id");
+        Assertions.assertNotNull(configId);
+
+        HashMap configData = GlobalConfigurationHelper.getGlobalConfigurationById(requestSpec, responseSpec, configId.toString());
+        Assertions.assertNotNull(configData);
+
+        Boolean enabled = (Boolean) globalConfig.get(3).get("enabled");
+
+        if (!enabled) {
+            enabled = true;
+            GlobalConfigurationHelper.updateEnabledFlagForGlobalConfiguration(requestSpec, responseSpec, configId, enabled);
+        }
+
+        holidayId = HolidayHelper.activateHolidays(requestSpec, responseSpec, holidayId.toString());
+        Assertions.assertNotNull(holidayId);
+
+        HashMap holidayData = HolidayHelper.getHolidayById(requestSpec, responseSpec, holidayId.toString());
+        LinkedHashMap repaymentScheduleHashMap = JsonPath.from(loanDetails).get("repaymentSchedule");
+        ArrayList<LinkedHashMap> periods = (ArrayList<LinkedHashMap>) repaymentScheduleHashMap.get("periods");
+        String JobName = "Apply Holidays To Loans";
+
+        this.schedulerJobHelper.executeAndAwaitJob(JobName);
+
+        // Loan Repayment Schedule After Apply Holidays To Loans
+        loanDetails = this.loanTransactionHelper.getLoanDetails(requestSpec, responseSpec, loanID);
+        repaymentScheduleHashMap = JsonPath.from(loanDetails).get("repaymentSchedule");
+        ArrayList<LinkedHashMap> periodsAfterRescheduleApplied = (ArrayList<LinkedHashMap>) repaymentScheduleHashMap.get("periods");
+        ArrayList<Integer> dateToApplyHolidays = null;
+
+        int periodsLength = periodsAfterRescheduleApplied.size();
+        for (int i = 0; i < periodsLength; i++) {
+            LinkedHashMap periodRescheduled = periodsAfterRescheduleApplied.get(i);
+            ArrayList<Integer> fromDate = (ArrayList<Integer>) periodRescheduled.get("fromDate");
+            LinkedHashMap periodBeforeRescheduled = periodsAfterRescheduleApplied.get(i);
+            ArrayList<Integer> dueDateBeforeRescheduled = (ArrayList<Integer>) periodBeforeRescheduled.get("dueDate");
+            if (fromDate != null && Objects.equals(fromDate.get(1), dueDateBeforeRescheduled.get(1))) {
+                dateToApplyHolidays = fromDate;
+            }
+            Assertions.assertNotNull(dateToApplyHolidays);
```

Comment:
its still not perfect...

i was wondering whether you could test something like this:

disburse the loan on a date (example: 2023.01.01)
Assert all the periods due date.
do the reschedule, (example: reschedule the 2nd period)
assert all the "new" periods due date:

the 1st period due date is not changed. 
the 2nd period due date was changed to the next one.
all the later periods was moved 


i hope it gives some insight! :)

Reply:
@adamsaghy  I tried to follow similar approach. Can you please review the latest commit?  I will squash the commits from next as I pushed these already

Reply:
But you did not follow at all... you are still calculating and comparing dates with a logic which might be correct but i am having hard time to read and understand. Since you are using a fixed date: 2023.01.01 where the loan is created and disbursed, the period from and due dates are fixed as well:
2023.02.01
2023.03.01
2023.04.01
etc.

Can you please write assertions which checks whether the above dates are the ones that got used and after the holiday reschedule what are the new dates. Example:

Number of repayments: 3
Period lenght: 1 month

**Before holiday reschedule**
// 1st period
`assertEquals(LocalDate.of(2023,1,1), periods.get(0).getFromDate())`
`assertEquals(LocalDate.of(2023,2,1), periods.get(0).getDueDate())`
// 2nd period
`assertEquals(LocalDate.of(2023,2,1), periods.get(1).getFromDate())`
`assertEquals(LocalDate.of(2023,3,1), periods.get(1).getDueDate())`
// 3rd period
`assertEquals(LocalDate.of(2023,3,1), periods.get(2).getFromDate())`
`assertEquals(LocalDate.of(2023,4,1), periods.get(2).getDueDate())`

**After holiday reschedule**

**Reschedule from 1st of Feb**
// 1st period
`assertEquals(LocalDate.of(2023,1,1), periods.get(0).getFromDate())`
`assertEquals(LocalDate.of(2023,2,1), periods.get(0).getDueDate())`
// 2nd period (got rescheduled)
`assertEquals(LocalDate.of(2023,2,1), periods.get(1).getFromDate())`
`assertEquals(LocalDate.of(2023,4,1), periods.get(1).getDueDate())`
// 3rd period (got rescheduled)
`assertEquals(LocalDate.of(2023,4,1), periods.get(2).getFromDate())`
`assertEquals(LocalDate.of(2023,5,1), periods.get(2).getDueDate())`


This is easy to read and easy to understand what is happening

Reply:
@adamsaghy  Hardcoding these values will not work. On every testing actual rescheduled dates will shift to next dates. If we hard code these values in the first run it will get succeeded but in the next run due dates will get changed but hard coded values will remain same.

Reply:
@2vinodhkumar  you are creating a brand new loan with each execution and checking only the created loan periods. so i dont think it will be different for each execution at all.

Reply:
@adamsaghy I have tested this from UI/API , seems working as expected. From integration tests, All the reschedules are getting updated though the due date is before the holiday date. Let me recheck with different test cases.

Reply:
I understand what you say, but with each test you are creating a brand new loan and that loan will be rescheduled just once (hence it is brand new). And that loan periods are checked, so it should be always the same. Feel free to test it ;)

Reply:
@adamsaghy  I have pushed new test cases and tested this working as expected. Let me know your comments once review it.